_posts/2023-09-10-python-multiple-arithmetic-operators.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "The surprising behavior of multiple + and - operators in Python"
+header:
+  image: /assets/images/headers/2023-09-28_pythons-weird-arithmetic.png
+  image_description: "An example of Python's weird arithmetic"
+  teaser: /assets/images/teasers/2023-09-28_pythons-weird-arithmetic.png
+excerpt: Let's explore this strange quirk of the Python language and learn from it.
+---
+
+All programming languages have some quirks and strange behavior.
+Python is certainly no exception.
+I was however quite surprised recently when I found out that Python allows multiple `+` and `-` in a row.
+Let's explore this behavior and learn about order of operation from it.
+
+# Use as many `+` and `-` together as you want
+
+I do not know the purpose of this feature but in Python you can write multiple `+` or `-` operators in a row.
+
+```python
+>>> 1 +++++++ 1
+2
+>>> 1 ++++++++++++++++++++ 1
+2
+>>> 1 ------- 1
+0
+>>> 1 ------ 1
+2
+```
+
+The operators behave as if they were shorthands for `* (+1)` and `* (-1)`.
+I do not know of any other language that behaves like this.
+I also do not see any purpose to it.
+
+The only use case that I can think of is if you need to ask "help" from a fellow developer to deal with your annoying manager:
+
+```python
+my_need = 1 --- +++ --- 5
+```
+
+There is always someone familiar with morse code in movies so it might work.
+
+# Using multiple `-` with `//` is weird
```

Comment:
Nice! I like that you're building on it! Base case and then "hey, I wonder how it works with ___"